### PR TITLE
Fix crash in `process.ProcessName` due to race.

### LINKED ISSUE #6572: Nuget fails with TFS credential provider
In about 90%+ of the cases when using the nuget task in a TFS 2017 release definition, using version 4.0.0, while also allowing access to the OAuth token, nuget failes with an InvalidOperationException, stating that the process has exited.  Following the stack trace, I found that it occurs in the PluginCredentialProvider class in an anonymous delegate defined in the Execute method.  A call to process.ProcessName is causing the exception.  I see that a workaround was added for #4004, such that this property is only accessed if the process has not yet exited.  Given this case, that means that either process.HasExited or process.ProcessName is lying as to whether the process has already exited, or the process happens to exit between the call to process.HasExited and the call to process.ProcessName.

The reason I raise this as an issue is that this condition happens so often that when it does work properly appears to be the exception rather than the rule.

### PR DESCRIPTION
**Pull request by CosminRadu**
Details:
The code in PluginCredentailProvider.Execute() suffers from a race condition between the call to Process.HasExited returning false and the following call to Process.ProcessName. The fix has been proposed by TheJayMann (https://github.com/TheJayMann) in his issue report: https://github.com/NuGet/Home/issues/6572.

## Bug
Fixes: https://github.com/NuGet/Home/issues/6572  
Regression: No  
If Regression then when did it last work:   
If Regression then how are we preventing it in future:   

## Fix
Details: The code in PluginCredentailProvider.Execute() suffers from a race condition between the call to Process.HasExited returning false and the following call to Process.ProcessName. The fix has been proposed by TheJayMann (https://github.com/TheJayMann) in his issue report: https://github.com/NuGet/Home/issues/6572. 

## Testing/Validation
Tests Added: No  
Reason for not adding tests: Race condition
Validation done: Local validation.

### CODE REVIEW

**Comment by nkolev92** [comment]
This is the old plugin model. 
We have replaced that plugin model with a new one in 4.8, and we are considering deprecating the old one in 5.0. 

My vote would be to not merge this for now. 
We can loop back once we have a decision on the old plugin model. 

https://docs.microsoft.com/en-us/nuget/reference/extensibility/nuget-cross-platform-authentication-plugin

//cc @rrelyea @anangaur

**Review by jainaashish — APPROVED** [review]

**Comment by jainaashish** [comment]
Change looks safe to take which could help customers still working with old plugin. If we're not going to deprecate old plugin, then we should take it.

@rrelyea  we should decide either way and accordingly complete this PR.

**Comment by rrelyea** [comment]
Let's hinge this decision on a 5.0 deprecation or not decision. Let's wait for @anangaur to return from vacation.

**Comment by nkolev92** [comment]
Deprecation issue: 
https://github.com/NuGet/Home/issues/7586

**Comment by nkolev92** [comment]
Hey @CosminRadu apologies for the delay. 

Can you please rebase so we can run the tests and merge?

**Review by nkolev92 — APPROVED** [review]

**Comment by nkolev92** [comment]
@CosminRadu Can you please sign the contributor license agreement? 
We cannot merge without that.

**Comment by nkolev92** [comment]
Closing this as we cannot merge without a signed CLA.

**Pull request closed without merging**
